src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -359,21 +359,21 @@ FixedwingPositionControl::airspeed_poll()
 {
 	bool airspeed_valid = _airspeed_valid;
 
-	if (!_parameters.airspeed_disabled && _airspeed_sub.update()) {
+	if (!_parameters.airspeed_disabled && _airspeed_validated_sub.update()) {
 
-		const airspeed_s &as = _airspeed_sub.get();
+		const airspeed_validated_s &airspeed_validated = _airspeed_validated_sub.get();
+		_eas2tas = 1.0f; //this is the default value, taken in case of invalid airspeed
 
-		if (PX4_ISFINITE(as.indicated_airspeed_m_s)
-		    && PX4_ISFINITE(as.true_airspeed_m_s)
-		    && (as.indicated_airspeed_m_s > 0.0f)
-		    && !_vehicle_status.aspd_use_inhibit) {
+		if (PX4_ISFINITE(airspeed_validated.equivalent_airspeed_m_s)
```

Comment:
I think we should be able to drop these checks and trust our "validated airspeed".

Reply:
The way it works currently is that the airspeed validated is set to NAN inside the airspeed module if it is detected to be invalid (or user has disabled it by setting PRIMARY to -1). As I see it we either keep it this way and check for the finiteness of the airspeed_validated where it's used, or build in some logic in commander that sets a "airspeed_inhibited" flag in vehicle_status, or make this flag part of the airpseed_validated message (as it was at some point, but was removed because it then contains redundant information (setting it to NAN should be equal to "don't use airspeed)". 
I'd vote for bringing it in as it is now.

Reply:
That's fine as long as every consumer ends up with the same notion of airspeed validity.